drivers/bluetooth/hci/userchan.c

```diff
@@ -93,24 +93,70 @@ static bool is_hci_event_discardable(const uint8_t *evt_data)
 	}
 }
 
-static struct net_buf *get_rx(const uint8_t *buf)
+static struct net_buf *get_rx_evt(const uint8_t *data)
 {
-	bool discardable = false;
-	k_timeout_t timeout = K_FOREVER;
+	bool discardable = is_hci_event_discardable(data);
+	struct net_buf *buf;
+
+	do {
+		buf = bt_buf_get_evt(data[1], discardable, discardable ? K_NO_WAIT : K_SECONDS(10));
```

Comment:
Since `discardable` is basically a `const`, the timeout here could also be similarly defined as a constant, so consider adding something like `const k_timeout_t timeout = discardable ? K_NO_WAIT : K_SECONDS(10)` after `bool discardable = is_hci_event_discardable(data);` and then modify this to 
```suggestion
		buf = bt_buf_get_evt(data[1], discardable, timeout);
```

I realize that will make this be slightly different from the IPC version, so it's not a blocker

Reply:
ok, I'll update it to consts